lib/ps.js

```diff
@@ -0,0 +1,72 @@
+var exec = require('child_process').exec
+var os = require('os')
+var PLATFORM = os.platform()
+
+/**
+  * Get pid informations through ps command
+  * @param   {Number|Number[]}  pid
+  * @param   {Object}           options
+  * @return  {Function}         callback(err, stat)
+  * on os x skip headers with pcpu=,rss=
+  * on linux it could be --no-header
+  * on solaris 11 can't figure out a way to do this properly so...
+  */
+function ps (pid, options, done) {
+  if (!Array.isArray(pid)) {
+    pid = parseInt(pid, 10)
+  } else {
+    pid = pid.map(function (pid) {
+      return parseInt(pid, 10)
+    }).join(',')
+  }
+
+  var cmd = 'ps -o pcpu,rss,pid -p '
+
+  if (options.advanced) {
+    cmd = 'ps -o pcpu,rss,pid,time -p '
+
+    if (PLATFORM === 'aix') {
+      cmd = 'ps -o pcpu,rssize,pid,time -p '
+    }
+  } else if (PLATFORM === 'aix') {
+    cmd = 'ps -o pcpu,rssize,pid -p ' // this one could work on other platforms see AIX section in man ps
+  }
+
+  exec(cmd + pid, function (err, stdout, stderr) {
+    if (err) {
+      return done(err, null)
+    }
+
+    stdout = stdout.split(os.EOL)
+
+    var statistics = []
+    for (var i = 1; i < stdout.length; i++) {
+      var line = stdout[i].replace(/^\s+/, '').replace(/\s\s+/g, ' ').split(' ')
+      var time = 0
+      if (options.advanced) {
+        // Example output 7-22:43:36
+        var tmp = line[3].split(':')
+        var days = 0
+        if (tmp[0].indexOf('-')) {
+          var daysAndHours = tmp[0].split('-')
+          tmp[0] = daysAndHours[1]
+          days = daysAndHours[0]
+        }
+
+        time = (days * 86400 + tmp[0] * 3600 + tmp[1] * 60 + tmp[2]) * 1000
+      }
+
+      statistics[i - 1] = {
+        cpu: parseFloat(line[0].replace(',', '.')),
+        memory: parseFloat(line[1]) * 1024,
+        pid: parseInt(line[2], 0),
+        time: time, // time in ms
+        start: new Date(Date.now() - time)
```

Comment:
You need to move
`Date.now()`
before 
`stdout = stdout.split(os.EOL)`

To have better precision.

Reply:
nice one yes